src/platform/platform_winkernel.c

```diff
@@ -250,11 +251,15 @@ QuicEtwCallback(
     UNREFERENCED_PARAMETER(MatchAllKeyword);
     UNREFERENCED_PARAMETER(FilterData);
 
+    if (!QuicTraceRundownCallback) {
+        return;
+    }
+
     switch(ControlCode) {
     case EVENT_CONTROL_CODE_ENABLE_PROVIDER:
     case EVENT_CONTROL_CODE_CAPTURE_STATE:
         if (CallbackContext == &MICROSOFT_MSQUIC_PROVIDER_Context) {
-            QuicTraceRundown();
+            QuicTraceRundownCallback();
```

Comment:
Grab a local copy of this function pointer and check that for null rather then checking the global for null then calling. To reduce the risk an outside caller sets the pointer to null after the check but before the call.

Reply:
I _could_ do that, but I don't think we need to worry about that added complexity. I'd prefer to keep the design of "this must be set (or not) before platform load and never written to again"

Reply:
Because if we did what you're suggesting that seems to encourage changing this at runtime. But there is no synchronization, so the setter might still get a callback after they clear this.

Reply:
Then I think instead of exposing the function pointer directly, we should expose a function that only ever allows the variable to be set once, and not be allowed to set it to null.

Reply:
I think I prefer the current model. A function then introduces the complexity around when do I call it? I'd rather just get this in so we can unblock the static linking PR. We can iterate more on this afterwards.

Reply:
As long as we make a Nickel bug to fix that then I'm ok with that. But its a correctness bug, so it does needs to be fixed before the next release.